source/Engine.cpp

```diff
@@ -1336,14 +1338,35 @@ void Engine::EnterSystem()
 
 	// Clear any active weather events
 	activeWeather.clear();
+
+	// If any fleets have an initial spawn count, spawn them.
+	for(const auto &fleet : system->Fleets())
+		if(fleet.InitialCount() > 0 && !fleet.Category().empty())
+		{
+			size_t toPlace = FleetPlacementLimit(fleet, 0, true);
+			for(size_t i = 0; i < toPlace ; ++i)
+			{
+				fleetShips.clear();
+				fleet.Get()->Place(*system, fleetShips, true);
+				AddSpawnedFleet(fleet);
+			}
+		}
+
 	// Place five seconds worth of fleets and weather events. Check for
 	// undefined fleets by not trying to create anything with no
 	// government set.
 	for(int i = 0; i < 5; ++i)
 	{
 		for(const auto &fleet : system->Fleets())
-			if(fleet.Get()->GetGovernment() && Random::Int(fleet.Period()) < 60)
-				fleet.Get()->Place(*system, newShips);
+			// Skip fleets that don't want to spawn on system entry,
+			// or fleets whose limits have already been reached.
+			if(!fleet.GetFlags(Fleet::SKIP_SYSTEM_ENTRY) &&
+				FleetPlacementLimit(fleet, 60, true))
```

Comment:
here we have use of the magic value 60, see below, https://github.com/endless-sky/endless-sky/pull/8082/files#r1855209017